manifests/init.pp

```diff
@@ -154,6 +154,7 @@
   Boolean $mime_types_preserve_defaults                      = false,
   Optional[String] $repo_release                             = undef,
   $passenger_package_ensure                                  = 'present',
+  $repo_source                                               = undef,
```

Comment:
Can you make this an optional http url type?
```suggestion
  Optional[Stdlib::HTTPUrl] $repo_source                     = undef,
```

https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/httpurl.pp . This requires at least stdlib 5.0.0